charms/kfp-profile-controller/tests/integration/test_charm.py

```diff
@@ -35,19 +44,181 @@ async def test_build_and_deploy(ops_test: OpsTest):
         "minio:object-storage",
     )
 
-    # TODO: Need a better test for checking everything is ok
+    # Deploy charms responsible for CRDs creation
+    await ops_test.model.deploy(
+        entity_url="kubeflow-profiles",
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        entity_url="metacontroller-operator",
+        trust=True,
+    )
+
     # Maybe: await ops_test.model.wait_for_idle(raise_on_error=False, raise_on_blocked=True) ?
-    await ops_test.model.wait_for_idle(timeout=60 * 10)
-
-
-# TODO: Real testing requires:
-#  * kubeflow-profile controller, or something that will make namespaces/service accounts in a
-#    similar fashion (some resources deployed to each namespace by kfp-profile-controller use the
-#    `default-editor` service account, which is created by the main kubeflow-profile controller)
-#  * This charm deploying successfully is not a full test of function.  We need to create a new
-#    tracked namespace (namespace that has label pipelines.kubeflow.org/enabled=true) and ensure
-#    all expected resources are deployed and come up successfully (eg: if service account is not
-#    found, deployments will exist but pods will never be "ready")
-#  * check whether secrets and other things get updated based on the config sent to sync.py/kfppc
-#  * could create namespaces with the right pipelines label to trigger the sync, but also need to
-#    create a dummy poddefaults CRD too
+    await ops_test.model.wait_for_idle(status="active", timeout=60 * 10)
+
+
+async def test_status(ops_test: OpsTest):
+    charms = ("kubeflow-profiles", "metacontroller-operator", "kfp-profile-controller")
+    for charm in charms:
+        assert ops_test.model.applications[charm].units[0].workload_status == "active"
+
+
+async def test_profile_creation(lightkube_client, profile):
+    # Test whether a namespace was created for this profile
+    profile_name = profile
+    validate_profile_resources(lightkube_client, profile_name)
```

Comment:
Small nit similar to that above.  The naming here makes me think you're validating something about the `Profile`, not about the resources you're creating for the profile.  If you can think of a better name that's great, otherwise leave as is